### PR TITLE
Speedup Qwen2VLImageProcessor

### PR DESCRIPTION
**Pull request by lgeiger**
# What does this PR do?

`Qwen2VLImageProcessor` currently spends a significant amount copying data during preprocessing:
Screenshot 2026-04-30 at 11 16 28

For images the amount of data copies can be reduced which speedsup preprocessing by **22% end2end**:
```python
import numpy as np
from PIL import Image
from transformers import Qwen2VLImageProcessor

rng = np.random.default_rng(0)
processor = Qwen2VLImageProcessor()
images = [Image.fromarray(rng.integers(0, 255, (1024, 1024, 3), dtype=np.uint8)) for _ in range(8)]

%timeit processor(images)
```
```
main:    33 ms ± 399 μs per loop (mean ± std. dev. of 7 runs, 10 loops each)
this PR: 25.7 ms ± 498 μs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

## Code Agent Policy

The Transformers repo is currently being overwhelmed by a large number of PRs and issue comments written by
code agents. We are currently bottlenecked by our ability to review and respond to them. As a result, 
**we ask that new users do not submit pure code agent PRs** at this time. 
You may use code agents in drafting or to help you diagnose issues. We'd also ask autonomous "OpenClaw"-like agents
not to open any PRs or issues for the moment.

PRs that appear to be fully agent-written will probably be closed without review, and we may block users who do this
repeatedly or maliciously. 

This is a rapidly-evolving situation that's causing significant shockwaves in the open-source community. As a result, 
this policy is likely to be updated regularly in the near future. For more information, please read [`CONTRIBUTING.md`](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md).

- [x] I confirm that this is not a pure code agent PR.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

/cc @zucchini-nlp

Best to review the diff with hidden whitespace

### CODE REVIEW

File: src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py [hunk 1]
```diff
@@ -193,31 +193,50 @@ def _preprocess(
             )
             if patches.ndim == 4:
                 patches = patches.unsqueeze(1)
-            if patches.shape[1] % temporal_patch_size != 0:
-                repeats = patches[:, -1:].repeat(1, temporal_patch_size - 1, 1, 1, 1)
-                patches = torch.cat([patches, repeats], dim=1)
-            batch_size, grid_t, channel = patches.shape[:3]
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py:199]: we should have deleted this tbh, since we don't accept `videos` as arg anymore in v5. So the frame count will always be 1

File: src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py [hunk 2]
```diff
@@ -193,31 +193,50 @@ def _preprocess(
...
+            if frames == 1:
+                grid_t = 1
+                patches = patches.reshape(
+                    batch_size,
+                    channel,
+                    grid_h // merge_size,
+                    merge_size,
+                    patch_size,
+                    grid_w // merge_size,
+                    merge_size,
```
Comment (zucchini-nlp) [inline src/transformers/models/qwen2_vl/image_processing_qwen2_vl.py]: and then we leave code for this branching only

**Review by zucchini-nlp — COMMENTED** [review]
Oh, and also, can you check if this pattern persists in other models like qwen3-vl, ernie, video-llama3?

**Comment by lgeiger** [comment]
@zucchini-nlp Thanks for the fast review! I've removed the video handling in a72feced887c4bd35a110a11b42dad7d006e758c and also adjusted ernie and video-llama3 accordingly. qwen3-vl re-uses the qwen2-vl processing so nothing to update there.

**Comment by zucchini-nlp** [comment]
run-slow: ernie4_5_vl_moe, glm_image, qwen2_vl, video_llama_3

**Review by zucchini-nlp — APPROVED** [review]
Thanks a lot @lgeiger , lgtm, merging

**Comment by lgeiger** [comment]
I just realised that I messed up the changes in video llama 3, d19ace6cde8b49b95262dc47d5260c99e8551443 should fix this.

**Comment by zucchini-nlp** [comment]
The temporal patch size in video-llama seems to be an artifact copied from qwen, and is always 1. So shouldn't be a diff whether we expand a dim, or not :)

**Comment by lgeiger** [comment]
> The temporal patch size in video-llama seems to be an artifact copied from qwen, and is always 1. So shouldn't be a diff whether we expand a dim, or not :)

Ah OK, got it. Do you prefer keeping it as is or should I revert d19ace6cde8b49b95262dc47d5260c99e8551443 again?

**Comment by zucchini-nlp** [comment]
Nah, it is fine, let's keep for complete BC since we never know what users might be doing in custom code
